builder/assignmentEnv.go

```diff
@@ -28,15 +34,22 @@ type baseEnvImage interface {
 	undoBuild() error
 }
 
+// assignmentEnvironment struct type holds parameters required
```

Comment:
maybe change the name to `assignmentEnvironmentImageBuilder`. With this the name of the corresponding source file will also have to change.

```go
// assignmentEnvironmentImageBuilder is used to build an image with an environment
// that satisfies all the language and library dependencies of an assignment.
//
// The image is built in phases to allow for rollback in situations where the execution of a phase fails.
// Once the configuration passes a verification phase, a Dockerfile is created and
// the docker client (here - https://github.com/moby/moby/tree/master/client) is used to generate the image.
// The image is then pushed to docker hub repository, if required.
```

Reply:
Okay